otelcol/flags.go

```diff
@@ -50,9 +51,12 @@ func flags(reg *featuregate.Registry) *flag.FlagSet {
 			return nil
 		})
 
-	flagSet.Var(featuregate.NewFlag(reg), featureGatesFlag,
-		"Comma-delimited list of feature gate identifiers. Prefix with '-' to disable the feature. '+' or no prefix will enable the feature.")
+	gatesFlag, strictFlag := featuregate.NewFlags(reg)
+	flagSet.Var(strictFlag, featureGatesStrictFlag,
```

Comment:
Why is this a `Var` instead of a boolean flag?

Reply:
Because a bool flag does not take a function, and instead returns a bool pointer, depriving me from the possibility to react when the flag is set.

Reply:
Right, thanks! That makes sense, but I think we should follow what is stated on https://pkg.go.dev/flag#Value:

> If a Value has an IsBoolFlag() bool method returning true, the command-line parser makes -name equivalent to -name=true rather than using the next command-line argument.

And add `IsBoolFlag` to `strictFlagValue`

Reply:
Cool, I'll add it. Thanks for teaching me this.